test/interpreter/InterpretConst.spec.ts

```diff
@@ -0,0 +1,52 @@
+
+import { CommonModel } from '../../src/models/CommonModel';
+import interpretConst from '../../src/interpreter/InterpretConst';
+import {inferTypeFromValue} from '../../src/interpreter/Utils';
+jest.mock('../../src/interpreter/Utils');
+describe('Interpretation of const', function() {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+  afterAll(() => {
+    jest.restoreAllMocks();
+  });
+  test('should not do anything for boolean schemas', function() {
+    const model = new CommonModel();
+    const schema: any = true;
+    interpretConst(schema, model);
+    expect(model.type).toBeUndefined();
+    expect(model.enum).toBeUndefined();
+  });
+  test('should not do anything if schema does not contain const', function() {
+    const model = new CommonModel();
+    const schema: any = { type: 'string'};
+    interpretConst(schema, model);
+    expect(model.type).toBeUndefined();
+    expect(model.enum).toBeUndefined();
+  });
+  test('should not infer type from const if schema have type', function() {
+    const model = new CommonModel();
+    const schema: any = { type: 'string', const: 'test'};
+    interpretConst(schema, model);
+    expect(model.type).toBeUndefined();
+    expect(model.enum).toEqual([schema.const]);
+  });
+  test('should infer type and enum', function() {
+    (inferTypeFromValue as jest.Mock).mockReturnValue("string");
+    const schema: any = { const: 'test'};
+    const model = new CommonModel();
+    interpretConst(schema, model);
+    expect(model.enum).toEqual([schema.const]);
+    expect(inferTypeFromValue).toHaveBeenNthCalledWith(1, 'test');
+    expect(model.type).toEqual("string");
+  });
+  test('should not infer unknown type', function() {
+    (inferTypeFromValue as jest.Mock).mockReturnValue(undefined);
+    const schema: any = { const: 'test'};
```

Comment:
You can use object `const: {}`

Reply:
Ok sorry 😅 Then you will have `object` type... So checking undefined is necessary?

Reply:
I can change it to `{}` if you think that makes it more clear, because of `(inferTypeFromValue as jest.Mock).mockReturnValue(undefined);` it does not really matter what the const is 🙂

Reply:
Yep, but even if you change to `object` you will infer `object` type. Probably `undefined` will never happen.

Reply:
In practice, not really no. In theory, it could

```js
const jsonSchema = {
    type: "string",
    const: () => {}
}
const generator = new TypeScriptGenerator({ modelType: 'interface' });
const interfaceModel = await generator.generate(jsonSchema);
```

Reply:
The test is purely there to ensure that the branch is covered 🙂